documentation/manual/src/main/asciidoc/en-US/modules/architecture.asciidoc

```diff
@@ -234,6 +234,31 @@ when the schema is changed as the schema is right in front of the developers' ey
 This is an intermediary model between the strictly typed relational model
 and the totally schema-less approach pushed by some NoSQL families.
 
+[[ogm-architecture-sequences]]
+
+=== Id generation using sequences
+
+You can use sequences with the following annotations:
+
+* [classname]`@SequenceGenerator`:
+  it will use native sequences if available
+* [classname]`@TableGenerator`:
+  it will emulate sequences storing the value in the most appropriate data structure;
+  for example a document in MongoDB or a node in Neo4j.
+
+Here's some things to keep in mind when dealing with sequence generation:
+
+* [classname]`@TableGenerator` is the fallback approach used when the underlying datastore does not
+  support native sequences generation.
+* If the datastore does not support atomic operations and does not support native sequences,
+  Hibernate OGM will throw an exception.
```

Comment:
This sounds worse than the reality. Could you clarify that the expcertion is thrown at bootstrap, and that it will suggest to use a different ID generation strategy for this dialect?